src/UI/Internal/FiltersHeaders.elm

```diff
@@ -287,6 +308,180 @@ singleTextFilterRender renderConfig applyMsg { fromFiltersMsg, index, label } ed
     editable
         |> Filters.editableWithDefault ""
         |> TextField.singlelineText editMsg label
+        |> TextField.withSize Size.extraSmall
         |> TextField.withWidth TextField.widthFull
         |> TextField.withOnEnterPressed applyMsg
         |> TextField.renderElement renderConfig
+
+
+multiTextFilterRender :
+    RenderConfig
+    -> msg
+    -> Config msg
+    -> Filters.Editable (Array String)
+    -> Element msg
+multiTextFilterRender renderConfig applyMsg { fromFiltersMsg, index, label } editableArr =
+    let
+        editMsg subIndex str =
+            fromFiltersMsg <| Filters.EditMultiText { column = index, field = subIndex, value = str }
+
+        single subIndex line =
```

Comment:
I would try to use more explcit names, avoiding too generic names, `single` can be a lot of things, so maybe `textField` or `singleTextField` something like this.